packages/ra-core/src/reducer/admin/references/oneToMany.ts

```diff
@@ -114,6 +138,45 @@ export const getReferencesByIds = (
     return Object.keys(references).length > 0 ? references : null;
 };
 
+const getRelatedReferences = (previousState, resource) =>
+    Object.keys(previousState).filter(key => key.includes(resource));
+
+const removeDeletedReference = (removedId: Identifier) => (
+    previousState,
+    key
+) => {
+    const hasReferenceToRemovedId = previousState[key].ids.includes(removedId);
+
+    if (!hasReferenceToRemovedId) {
+        return previousState;
+    }
+
+    return {
+        ...previousState,
+        [key]: {
+            ids: previousState[key].ids.filter(id => id !== removedId),
+            total: previousState[key].total - 1,
+        },
+    };
+};
+
+const removeDeletedReferences = (removedIds: Identifier[]) => (
+    previousState,
+    key
+) => {
+    const idsToRemove = previousState[key].ids.filter(id =>
```

Comment:
Ain't the total equal to the number of ids ?
If so, you could filter to get the kept id, and returns 
```js
{
    ids: keptIds,
    total: keptIds.length
}
```
If not the test should reflect that fact.

Reply:
Also refactor removeDeletedReference to use removeDeletedReferences.